crates/dataflow-jit/src/ir/expr.rs

```diff
@@ -0,0 +1,483 @@
+    Add,
+    Sub,
+    Mul,
+    Div,
+    Eq,
+    Neq,
+    LessThan,
+    GreaterThan,
+    LessThanOrEqual,
+    GreaterThanOrEqual,
+    And,
+    Or,
+    Xor,
+    Min,
+    Max,
+    // TODO: shr, shl, rem, mod, sqrt, rotl, rotr
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnaryOp {
+    value: RValue,
+    kind: UnaryOpKind,
+}
+
+impl UnaryOp {
+    pub fn new(value: RValue, kind: UnaryOpKind) -> Self {
+        Self { value, kind }
+    }
+
+    pub const fn value(&self) -> &RValue {
+        &self.value
+    }
+
+    pub const fn kind(&self) -> UnaryOpKind {
+        self.kind
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum UnaryOpKind {
+    Abs,
+    Neg,
+    Not,
+    Ceil,
+    Floor,
+    Trunc,
+    Sqrt,
+    CountOnes,
+    CountZeroes,
+    LeadingOnes,
+    LeadingZeroes,
+    TrailingOnes,
+    TrailingZeroes,
+    BitReverse,
+    ByteReverse,
+}
+
+/// Copies a value
+#[derive(Debug, Clone, PartialEq)]
+pub struct CopyVal {
+    /// The value to be copied
+    value: ExprId,
+    ty: ColumnType,
+}
+
+impl CopyVal {
+    pub fn new(value: ExprId, ty: ColumnType) -> Self {
+        Self { value, ty }
+    }
+
+    pub const fn value(&self) -> ExprId {
+        self.value
+    }
+
+    pub const fn ty(&self) -> ColumnType {
+        self.ty
+    }
+}
+
+/// Extract a value from a row
+#[derive(Debug, Clone, PartialEq)]
+pub struct Load {
+    /// The row to extract from
+    source: ExprId,
+    /// The index of the column to extract from
+    column: usize,
+}
+
+impl Load {
+    pub fn new(target: ExprId, column: usize) -> Self {
+        Self {
+            source: target,
+            column,
+        }
+    }
+
+    pub const fn source(&self) -> ExprId {
+        self.source
+    }
+
+    pub const fn column(&self) -> usize {
+        self.column
+    }
+}
+
+/// Insert a value into a row
+#[derive(Debug, Clone, PartialEq)]
+pub struct Store {
+    /// The row to insert into
+    target: ExprId,
+    /// The index of the column to insert into
+    column: usize,
+    /// The value being inserted
+    value: RValue,
+}
+
+impl Store {
+    pub fn new(target: ExprId, column: usize, value: RValue) -> Self {
+        Self {
+            target,
+            column,
+            value,
+        }
+    }
+
+    pub const fn target(&self) -> ExprId {
+        self.target
+    }
+
+    pub const fn column(&self) -> usize {
+        self.column
+    }
+
+    pub const fn value(&self) -> &RValue {
+        &self.value
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct IsNull {
+    value: ExprId,
+    row: usize,
+}
+
+impl IsNull {
+    pub fn new(value: ExprId, row: usize) -> Self {
+        Self { value, row }
+    }
+
+    pub const fn value(&self) -> ExprId {
+        self.value
+    }
+
+    pub const fn row(&self) -> usize {
+        self.row
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct SetNull {
+    target: ExprId,
+    row: usize,
+    is_null: RValue,
```

Comment:
and this one?

Reply:
`row` is the same as the last one, `is_null` is the boolean value that's stored as the given column's null flag